### PR TITLE
feat: Add Validate Form

### PR DESCRIPTION
**Pull request by moshoodabdullahi**
# This PR adds the Portfolio Project Validate Form following the design and layout in the [Assignment Link](https://github.com/bolah2009/curriculum-html-css/blob/main/portfolio/9_validate_form.md) and the [Figma Design](https://www.figma.com/file/LKueaSlr8Kc2yMKeZkKgtm/portfolio_project?type=design&node-id=48-2228&mode=design&t=SrlYt6mk7ReAfYmd-0)

This milestone is used to implement a client-side validation of the data in the contact form before submission. and when the user submits the form, it checks if the email is in lowercase.

### CODE REVIEW

File: js/validate.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+const form = document.getElementById('my-form-button');
```
Comment (Yinkusadel) [inline js/validate.js]: first i noticed is i can submit your form without filling anything, lets fix that also the whole form validation should rely on js

**Review by Yinkusadel — CHANGES_REQUESTED** [review]
please make the changes

File: js/validate.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+const form = document.querySelector('#my-form');
```
Comment (Yinkusadel) [inline js/validate.js:1]: i noticed when there is an error it clears all the inputs then state the error after it has cleared all inputs lets fix that

File: js/validate.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+async function handleSubmit(event) {
+  event.preventDefault();
+
+  const isEmailValid = emailInput.value === emailInput.value.toLowerCase();
+
+  if (isEmailValid) {
+    errorMessage.textContent = '';
+
+    console.log('Form submitted successfully!');
```
Comment (Yinkusadel) [inline js/validate.js]: also why are we using console.log??

File: js/validate.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    headers: {
+      Accept: 'application/json',
+    },
+  })
+    .then((response) => {
+      if (response.ok) {
+        status.innerHTML = 'Thanks for your submission!';
+        form.reset();
+      } else {
+        response.json().then((data) => {
```
Comment (Yinkusadel) [inline js/validate.js:31]: also based on your logic here i dont think this is what you want in all cases,,you have an error here and it still says thanks for your submission whereas the form shouldnt submit at all if there is an error

image

**Review by Yinkusadel — CHANGES_REQUESTED** [review]
please work on those changes

**Review by Yinkusadel — DISMISSED** [review]
nice work

File: js/validate.js (hunk 4 above)
Comment (moshoodabdullahi) [reply js/validate.js:31]: Done

File: js/validate.js (hunk 3 above)
Comment (moshoodabdullahi) [reply js/validate.js]: Done

File: js/validate.js (hunk 2 above)
Comment (moshoodabdullahi) [reply js/validate.js:1]: Done

**Review by teckcipher — APPROVED** [review]

**Review by Yinkusadel — APPROVED** [review]
nice work

**Pull request merged**